skpro/distributions/discrete_truncated.py

```diff
@@ -0,0 +1,125 @@
+"""Left Truncated Discrete Distribution."""
+
+import numpy as np
+
+from skpro.distributions.base import BaseDistribution
+
+
+# TODO: Implement a general discrete truncation and let this inherit from it
+class LeftTruncatedDiscrete(BaseDistribution):
+    r"""A left truncated discrete distribution _not_ including the lower bound.
+
+    Assume that the base distribution is a discrete distribution with a support of
+    non-negative integers (e.g., Poisson, Binomial, etc.). This distribution samples
+    from the base distribution but excludes the values below a specified lower bound.
+    Mathematically, it can be expressed as:
+
+    .. math::
+        Y \sim f(y \vert y > \lambda) = \frac{f(y)}{1 - F(\lambda)},
+
+    where :math:`\lambda` is the lower bound, and :math:`f(y)` is the probability mass
+    function.
+
+    Parameters
+    ----------
+    distribution : BaseDistribution
+        The base discrete distribution from which to sample.
+
+    lower_bound : int
+        The lower bound below which values are truncated (excluded from sampling).
+
+    """
+
+    _tags = {
+        "capabilities:approx": ["energy", "pmf", "cdf"],
+        "capabilities:exact": ["ppf", "mean", "var", "log_pmf"],
+        "distr:measuretype": "discrete",
+        "distr:paramtype": "parametric",
+        "broadcast_init": "on",
+    }
+
+    def __init__(
+        self, distribution: BaseDistribution, lower_bound: int, index=None, columns=None
+    ):
+        assert distribution._tags["distr:measuretype"] == "discrete", ""
+
+        self.distribution = distribution
+        self.lower_bound = lower_bound
+
+        super().__init__(index=index, columns=columns)
+
+    def _log_pmf(self, x):
+        is_invalid = x <= self.lower_bound
+
+        log_prob_base = self.distribution.log_pmf(x)
+
+        log_prob_at_zero = self.distribution.log_pmf(self.lower_bound)
+        log_normalizer = np.log1p(-np.exp(log_prob_at_zero))
+
+        log_prob_truncated = log_prob_base - log_normalizer
+
+        return np.where(is_invalid, -np.inf, log_prob_truncated)
+
+    def _ppf(self, p):
+        low_cdf = self.distribution.cdf(self.lower_bound)
+        normalizer = 1.0 - low_cdf
+        x = low_cdf + normalizer * p
```

Comment:
I do not think this formula is correct. If `p=1` the result is `1`, which is in general false.